Avoid an ambiguous std::isnan overload for native _Float16 in C++17.

Convert to float explicitly, as the unary microkernel tester already does.

--- a/test/vbinary-microkernel-tester.cc
+++ b/test/vbinary-microkernel-tester.cc
@@ -64,7 +64,7 @@
     // Verify results.
     for (size_t i = 0; i < batch_size(); i++) {
-      if (std::isnan(y_ref[i])) {
+      if (std::isnan(static_cast<float>(y_ref[i]))) {
         // TODO: We could check if y[i] is NaN, but not all our kernels do this.
       } else {
         ASSERT_NEAR(
             y[i], y_ref[i],
